warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/handler/edge/evaluation/EdgePreconditionArithmetic.java

```diff
@@ -309,12 +312,109 @@ private void addMatchingGroup(Object o) {
         }
     }
 
+    private void addExcludedGroup(Object o) {
+        if (o instanceof EventFieldValueTuple) {
+            String fieldName = EventFieldValueTuple.getFieldName(o);
+            String commonality = EventField.getGroup(fieldName);
+            String group = EventField.getSubgroup(fieldName);
+            Set<String> groups = excludedGroups.get(commonality);
+            if (groups == null) {
+                groups = new HashSet<>();
+                groups.add(group);
+                excludedGroups.put(commonality, groups);
+            } else
+                groups.add(group);
+        }
+    }
+
     public Map<String,Set<String>> getMatchingGroups() {
         return matchingGroups;
     }
 
+    public Map<String,Set<String>> getExcludedGroups() {
+        return excludedGroups;
+    }
+
     public void clearMatchingGroups() {
         matchingGroups = new HashMap<>();
     }
 
+    public void clearExcludedGroups() {
+        excludedGroups = new HashMap<>();
+    }
+
+    public void clear() {
+        clearMatchingGroups();
+        clearExcludedGroups();
+    }
+
+    public Object notEquals(Object left, Object right) {
```

Comment:
How is this different than !equals(left, right) ?

Reply:
I need the logic to add the field and grouping info to exclusions.

Reply:
An alternative could be passing knowledge of the node being negated, like an isNegated flag, etc